pydantic/validators.py

```diff
@@ -210,6 +211,16 @@ def anystr_lower(v: 'StrBytes') -> 'StrBytes':
     return v.lower()
 
 
+def defaultdict_validator(v: Any) -> 'AnyDefaultdict':
+    if isinstance(v, defaultdict):
+        return v
+
+    try:
+        return defaultdict(None, v)
+    except (TypeError, ValueError):
+        raise errors.DictError()
```

Comment:
This won't work because you set the `default_factory` to `None`.
We should either forbid coercion to default dict or try to retrieve the default_factory.
Otherwise the issue you linked is not solved

```py
class Item(BaseModel):
    attrs: defaultdict = Field(default_factory=lambda: defaultdict(list))

item = Item()  # here item.attrs is a defaultdict(list)
item = Item({})  # here item.attrs is a defaultdict(None) so `item.attrs["a"].append(1)` will raise
```

We can maybe do something like
```py
def defaultdict_validator(v: Any, field: 'ModelField') -> 'AnyDefaultdict':
    if isinstance(v, defaultdict):
        return v

    the_defaultdict = field.default_factory()
    if not isinstance(the_defaultdict, defaultdict):
        raise ValueError('Could not infer `default_factory` for defaultdict')

    try:
        the_defaultdict.update(**v)
    except (TypeError, ValueError):
        raise errors.DictError()
    else:
        return the_defaultdict
```
🤷